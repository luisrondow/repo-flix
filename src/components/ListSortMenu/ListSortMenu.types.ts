export type ListSortMenuProps = {
  onSortOptionClick?: (prevValue: string) => void
}
