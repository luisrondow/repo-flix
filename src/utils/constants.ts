export const TECHS = ['Typescript', 'Javascript', 'Vue', 'Go', 'CSS', 'Node'] as const

export const BOOKMARKS_KEY = 'repoflix:bookmarks' as const
export const SELECTED_TECHS_KEY = 'repoflix:selected-techs' as const
