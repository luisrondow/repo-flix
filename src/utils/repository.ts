export type Repository = {
  id: string
  name: string
  url: string
  image: string
}
