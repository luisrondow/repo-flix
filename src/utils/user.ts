export type User = {
  uid: string
  email: string
  accessToken: string
  username: string
}
