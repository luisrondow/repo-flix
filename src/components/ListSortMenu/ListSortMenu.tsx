import { SORT_OPTIONS } from '../../utils/constants'
import { ListSortMenuProps } from './ListSortMenu.types'

const ListSortMenu = ({ activeSortOption, onSortOptionClick }: ListSortMenuProps) => {
  const handleSortOptionClick = (value: (typeof SORT_OPTIONS)[number]['value']) => {
    if (value === activeSortOption) return

    onSortOptionClick && onSortOptionClick(value)
  }

  return (
    <div className="absolute right-0 z-30 mr-2 w-64 sm:left-0">
      <li className="list-none bg-white">
        {SORT_OPTIONS.map(({ value, label }) => (
          <ul
            key={value}
            onClick={() => handleSortOptionClick(value)}
            className={`${
              activeSortOption === value && 'underline'
            } cursor-pointer px-2 py-1 font-semibold hover:bg-gray-light`}
          >
            Sort by {label}
          </ul>
        ))}
      </li>
    </div>
  )
}

export default ListSortMenu
