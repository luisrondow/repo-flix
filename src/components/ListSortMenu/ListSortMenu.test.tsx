import { render } from '@testing-library/react'
import ListSortMenu from './ListSortMenu'
import { SORT_OPTIONS } from '../../utils/constants'

describe('ListSortMenu', () => {
  const mockOnSortOptionClick = jest.fn()

  it('should render correctly', () => {
    const { container, getByText } = render(
      <ListSortMenu
        activeSortOption={SORT_OPTIONS[0].value}
        onSortOptionClick={mockOnSortOptionClick}
      />,
    )

    expect(getByText(`Sort by ${SORT_OPTIONS[0].label}`)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should fire onSortOptionClick correctly', () => {
    const { getByText } = render(
      <ListSortMenu
        activeSortOption={SORT_OPTIONS[0].value}
        onSortOptionClick={mockOnSortOptionClick}
      />,
    )

    getByText(`Sort by ${SORT_OPTIONS[1].label}`).click()

    expect(mockOnSortOptionClick).toHaveBeenCalledTimes(1)
    expect(mockOnSortOptionClick).toHaveBeenCalledWith(SORT_OPTIONS[1].value)
  })
})
