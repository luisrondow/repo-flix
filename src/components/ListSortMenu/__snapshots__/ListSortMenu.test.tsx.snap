// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSortMenu should render correctly 1`] = `
<div>
  <div
    class="absolute right-0 z-30 mr-2 w-64 sm:left-0"
  >
    <li
      class="list-none bg-white"
    >
      <ul
        class="underline cursor-pointer px-2 py-1 font-semibold hover:bg-gray-light"
      >
        Sort by 
        stars
      </ul>
      <ul
        class="false cursor-pointer px-2 py-1 font-semibold hover:bg-gray-light"
      >
        Sort by 
        forks
      </ul>
      <ul
        class="false cursor-pointer px-2 py-1 font-semibold hover:bg-gray-light"
      >
        Sort by 
        updated
      </ul>
      <ul
        class="false cursor-pointer px-2 py-1 font-semibold hover:bg-gray-light"
      >
        Sort by 
        help wanted issues
      </ul>
    </li>
  </div>
</div>
`;
